### PR TITLE
feat: implement `letter.lowercase` and `letter.uppercase`

### LINKED ISSUE #37: differentiate lowercase letter, uppercase letter, or combine them
### 🆒 Your use case

right now the function `letter` is include all the uppercase and lowercase letter, my use case is i only need the lowercase letter, and uppercase letter is prohibited.

### 🔍 Alternatives you've considered

_No response_

### ℹ️ Additional info

_No response_

### PR DESCRIPTION
**Pull request by ccjmne**
Pretty trivial implementation!

However, something is wonky in the changes to the typings tests I introduced in c2200a478fbceb25fcd88f274e064ab7068c9acb...
The tests were broken by introducing `lowercase` and `uppercase` properties to `letter`, and when I tried reproducing a minimal example of it on typescriptlang.org, of course there, an `Asdf<T> & { yes: 'no' }` **does** extend `Asdf<any>`. I couldn't put my finger on what the fix should be.

There might be a better way to handle them through savvy typing of the `extractRegExp` function, but I couldn't figure it out myself. 🙇‍♀️

Resolves #37.

### CODE REVIEW

**Review by danielroe — CHANGES_REQUESTED** [review]
The test issues you encountered are flagging that the type generation is broken at the moment.

(To be clear, this is the most difficult part of the PR for the reason you've encountered.)

We might consider rethinking the API. Maybe `letter.any`, `letter.lowercase`, `letter.uppercase` as three in parallel? wdyt @didavid61202

**Comment by didavid61202** [comment]
HI @danielroe, @ccjmne . I've made a PR #79 related to the type inferencing issue encounter in the PR. 
If #70 is merged, we can remove the type casting in test file and merge this PR 👍

File: test/inputs.test.ts [hunk 1]
```diff
@@ -20,6 +20,7 @@ import {
   linefeed,
   carriageReturn,
   charNotIn,
+  type Input,
```
Comment (danielroe) [inline test/inputs.test.ts:23]:
```suggestion
  charNotIn,
```

File: test/inputs.test.ts [hunk 2]
```diff
@@ -145,7 +156,11 @@ describe('inputs', () => {
     expect(not.whitespace.toString()).toMatchInlineSnapshot('"\\\\S"')
     expectTypeOf(extractRegExp(not.whitespace)).toEqualTypeOf<'\\S'>()
     expect(not.letter.toString()).toMatchInlineSnapshot('"[^a-zA-Z]"')
-    expectTypeOf(extractRegExp(not.letter)).toEqualTypeOf<'[^a-zA-Z]'>()
+    expectTypeOf(extractRegExp(not.letter as Input<'[^a-zA-Z]'>)).toEqualTypeOf<'[^a-zA-Z]'>()
+    expect(not.letter.lowercase.toString()).toMatchInlineSnapshot('"[^a-z]"')
+    expectTypeOf(extractRegExp(not.letter.lowercase)).toEqualTypeOf<'[^a-z]'>()
+    expect(not.letter.uppercase.toString()).toMatchInlineSnapshot('"[^A-Z]"')
+    expectTypeOf(extractRegExp(not.letter.uppercase)).toEqualTypeOf<'[^A-Z]'>()
```
Comment (danielroe) [inline test/inputs.test.ts:163]:
```suggestion
    expectTypeOf(extractRegExp(not.letter)).toEqualTypeOf<'[^a-zA-Z]'>()
    expect(not.letter.lowercase.toString()).toMatchInlineSnapshot('"[^a-z]"')
    expectTypeOf(extractRegExp(not.letter.lowercase)).toEqualTypeOf<'[^a-z]'>()
    expect(not.letter.uppercase.toString()).toMatchInlineSnapshot('"[^A-Z]"')
    expectTypeOf(extractRegExp(not.letter.uppercase)).toEqualTypeOf<'[^A-Z]'>()
```

**Review by danielroe — APPROVED** [review]
Thank you both! ❤️

**Comment by ccjmne** [comment]
This was really nice!
I should get off my butt and try to participate more in some open-source projects outside of Oktoberfest as well. 🤯

I learned some things, thought about some implementation details, read code that's not mine, discovered the Vite bundler, realised that pnpm is really getting popular...
All of that, by doing nearly nothing! 😂

Great experience 👍
Also, delightful interactions with you guys 😊

Cheers!

**Comment by didavid61202** [comment]
@ccjmne Thanks!
I'm also on the same track, trying to participate more in open-source projects, and have learned quite a lot from the best developers/maintainers like Daniel, Anthony, and Pooya Parsa (All from Nuxt team!). 
And I will highly recommend checking out [Nuxt repo](https://github.com/nuxt/framework) and [unjs repos](https://github.com/unjs) if you haven't 👍 

Thanks for your contribution! 💚 

Cheers! 🍻

**Pull request merged**